models/DataObject/Data/QuantityValue.php

```diff
@@ -58,9 +58,16 @@ public function __toString(): string
             }
         }
 
-        if ($this->getUnit() instanceof Unit) {
-            $translator = Pimcore::getContainer()->get('translator');
-            $value .= ' ' . $translator->trans($this->getUnit()->getAbbreviation(), [], 'admin');
+        $unit = $this->getUnit();
+        if ($unit instanceof Unit) {
+            if ($unit->getAbbreviation() === null) {
+                $unitAbbreviation = $unit->getId();
+            } else {
+                $translator = Pimcore::getContainer()->get('translator');
+                $unitAbbreviation = $translator->trans($unit->getAbbreviation(), [], 'admin');
+            }
+
+            $value .= ' ' . $unitAbbreviation;
```

Comment:
```suggestion
            $unitAbbreviation = $translator->trans($unit->getAbbreviation() ?? $unit->getId(), [], 'admin');
            $value .= ' ' . $unitAbbreviation;
```  

I'm wondering if this is also acceptable - simply fallback to unit's ID and let it be translatable too.

Reply:
I understand the suggestion and I like the clean approach. I'm just not sure if something like an ID should be translatable, even though it is a manually entered ID. The main reason for this change is to avoid a fatal error when no abbreviation was entered.

As to whether the space should be removed: I'll leave that up to a Pimcore developer to decide.

Reply:
@dolmit-tanel-paaro @kjkooistra-youwe  We have decided internal that the ID shouldn't be translated and also the removing of the space doesn't fix anything, so we can't do this in 11.5.  Let's merge this like it is at the moment. Thanks for your discussion